tests/graph/test_op.py

```diff
@@ -232,3 +232,55 @@ def perform(self, *_):
 
     x = pt.TensorType(dtype="float64", shape=(1,))("x")
     assert SomeOp()(x).type == pt.dvector
+
+
+def test_op_name():
+    op_name = "op_name"
+
+    class DummyType(Type):
+        def filter(self, data):
+            return data
+
+        def __eq__(self, other):
+            return isinstance(other, DummyType)
+
+        def __hash__(self):
+            return hash(DummyType)
+
+        def __repr__(self):
+            return "DummyType()"
+
+    def dummy_variable(name):
+        return Variable(DummyType(), None, None, name=name)
+
+    x = dummy_variable("x")
+
+    class SingleOutOp(Op):
+        def make_node(self, *inputs):
+            outputs = [dummy_variable("a")]
+            return Apply(self, list(inputs), outputs)
+
+        def perform(self, node, inputs, outputs):
+            raise NotImplementedError()
+
+    single_op = SingleOutOp()
+    res_single = single_op(x, name=op_name)
+    assert res_single.name == op_name
```

Comment:
This may be more readable. Also the name we are giving is not so much an `op_name` but a `var_name`. 
```suggestion
    res_single = single_op(x, name="test_name")
    assert res_single.name == "test_name"
```

Reply:
Also test that by default name is None?